<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Lecture 26: Genome assembly and annotation &mdash; bioinformatics@nau</title>
    
    <link rel="stylesheet" href="../../../static/haiku.css" type="text/css" />
    <link rel="stylesheet" href="../../../static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../../../',
        VERSION:     '',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../../../static/jquery.js"></script>
    <script type="text/javascript" src="../../../static/underscore.js"></script>
    <script type="text/javascript" src="../../../static/doctools.js"></script>
    <link rel="top" title="bioinformatics@nau" href="../../../index.html" />
    <link rel="up" title="Lecture outlines" href="index.html" />
    <link rel="next" title="Course schedule" href="../course_schedule.html" />
    <link rel="prev" title="Lecutre 25: Genome assembly" href="lecture25.html" /> 
  </head>
  <body>
      <div class="header"><h1 class="heading"><a href="../../../index.html">
          <span>bioinformatics@nau</span></a></h1>
        <h2 class="heading"><span>Lecture 26: Genome assembly and annotation</span></h2>
      </div>
      <div class="topnav">
      
        <p>
        «&#160;&#160;<a href="lecture25.html">Lecutre 25: Genome assembly</a>
        &#160;&#160;::&#160;&#160;
        <a class="uplink" href="../../../index.html">Contents</a>
        &#160;&#160;::&#160;&#160;
        <a href="../course_schedule.html">Course schedule</a>&#160;&#160;»
        </p>

      </div>
      <div class="content">
        
        
  <div class="section" id="lecture-26-genome-assembly-and-annotation">
<h1>Lecture 26: Genome assembly and annotation<a class="headerlink" href="#lecture-26-genome-assembly-and-annotation" title="Permalink to this headline">¶</a></h1>
<p>Redundant coverage in genome sequencing (in an <tt class="docutils literal"><span class="pre">Nx</span></tt> coverage genome sequencing project, each nucleotide is sequenced <tt class="docutils literal"><span class="pre">N</span></tt> times on average).</p>
<dl class="docutils">
<dt>Deeper coverage helps with:</dt>
<dd><ul class="first last simple">
<li>Error detection/correction: probability of an erroneous base when you have (say) eight overlapping calls of that base is very low.</li>
<li>Sequencing polymorphic alleles in diploid or polyploid genomes.</li>
</ul>
</dd>
</dl>
<p>Deeper coverage can&#8217;t help with resolving repeats that are longer than the read length. Paired reads can help with this.</p>
<p>Genome annotation: the process of identifying genes and other functional regions, and ideally assigning function to those regions. This is related to the process of ORF calling (see my lecture slides).</p>
<dl class="docutils">
<dt>Minimum genome sizes:</dt>
<dd><ul class="first last simple">
<li>Mycoplasma (a non-free-living bacterium) with around 500 genes. Through systematic mutatgenesis, estimated that 250-300 are necessary for a viable cell.</li>
<li>Bacillus subtilis has around 4100 genes, and similar results were found (around 270 genes were essential) but only when growing in a very rich media.</li>
<li>Most essential genes are typically present in bacteria, and ~70% are also present in archaea and eukaryotes.</li>
</ul>
</dd>
</dl>
<p>A recent comparison of genome assembly tools was performed as <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed?term=assemblathon">Assemblathon 1.0</a>. This was a shared evaluation project, where different tools were compared on their ability to assemble the same data. In this study the authors simulated a genome using <a class="reference external" href="http://www.drive5.com/evolver/">Evolver</a>.</p>
<p>To check out some existing genomes and associated metadata and annotations, follow these steps:</p>
<blockquote>
<div><ul class="simple">
<li>Go to the <a class="reference external" href="http://img.jgi.doe.gov/">Integrated Microbial Genomes (IMG site)</a>.</li>
<li>Click <tt class="docutils literal"><span class="pre">IMG</span></tt></li>
<li>Click <tt class="docutils literal"><span class="pre">Find</span> <span class="pre">Genomes</span></tt></li>
<li>Choose a genome and click it&#8217;s name</li>
<li>Click <tt class="docutils literal"><span class="pre">Browse</span> <span class="pre">Genome</span></tt></li>
<li>Click <tt class="docutils literal"><span class="pre">Download</span> <span class="pre">Gene</span> <span class="pre">Information</span></tt></li>
</ul>
</div></blockquote>
<p>N50 value: a metric of the quality of assembly: 50% of the reads are contained in contigs of length N or greater</p>
</div>


      </div>
      <div class="bottomnav">
      
        <p>
        «&#160;&#160;<a href="lecture25.html">Lecutre 25: Genome assembly</a>
        &#160;&#160;::&#160;&#160;
        <a class="uplink" href="../../../index.html">Contents</a>
        &#160;&#160;::&#160;&#160;
        <a href="../course_schedule.html">Course schedule</a>&#160;&#160;»
        </p>

      </div>

    <div class="footer">
        &copy; Copyright 2012, J Gregory Caporaso.
      Last updated on Apr 09, 2015.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.2.3.
    </div>
  </body>
</html>
